tests/test_tools_module.py

```diff
@@ -0,0 +1,784 @@
+        ),
+        patch(
+            "assisted_service_mcp.utils.auth.get_access_token",
+            return_value="test-access-token",
+        ),
+    ):
+        resp = await event_tools.host_events(lambda: "test-access-token", "cid", "hid")
+        assert json.loads(resp)["events"] == ["h1"]
+
+
+@pytest.mark.asyncio
+async def test_tool_cluster_iso_download_url_module() -> None:
+    from assisted_service_mcp.src.tools import download_tools
+    from assisted_service_mcp.src.mcp import AssistedServiceMCPServer
+
+    class _Presigned:
+        def __init__(self, url: str, expires_at: str | None = None) -> None:
+            self.url = url
+            self.expires_at = (
+                _dt.datetime.fromisoformat("2025-01-01T00:00:00+00:00")
+                if expires_at
+                else None
+            )
+
+    mock_client = Mock()
+    mock_client.list_infra_envs = AsyncMock(return_value=[{"id": "ie1"}])
+    mock_client.get_infra_env_download_url = AsyncMock(
+        return_value=_Presigned("https://u/iso", "2025-01-01T00:00:00Z")
+    )
+
+    AssistedServiceMCPServer()
+    with (
+        patch(
+            "assisted_service_mcp.src.tools.download_tools.InventoryClient",
+            return_value=mock_client,
+        ),
+        patch(
+            "assisted_service_mcp.utils.auth.get_access_token",
+            return_value="test-access-token",
+        ),
+    ):
+        resp = await download_tools.cluster_iso_download_url(
+            lambda: "test-access-token", "cid"
+        )
+        data = json.loads(resp)
+        assert data[0]["url"] == "https://u/iso"
+        assert data[0]["expires_at"] == "2025-01-01T00:00:00Z"
+
+
+@pytest.mark.asyncio
+async def test_tool_cluster_credentials_download_url_module() -> None:
+    from assisted_service_mcp.src.tools import download_tools
+    from assisted_service_mcp.src.mcp import AssistedServiceMCPServer
+
+    class _Presigned:
+        def __init__(self, url: str, expires_at: str | None = None) -> None:
+            self.url = url
+            self.expires_at = expires_at
+
+    mock_client = Mock()
+    mock_client.get_presigned_for_cluster_credentials = AsyncMock(
+        return_value=_Presigned("https://u/kubeconfig", None)
+    )
+
+    AssistedServiceMCPServer()
+    with (
+        patch(
+            "assisted_service_mcp.src.tools.download_tools.InventoryClient",
+            return_value=mock_client,
+        ),
+        patch(
+            "assisted_service_mcp.utils.auth.get_access_token",
+            return_value="test-access-token",
+        ),
+    ):
+        resp = await download_tools.cluster_credentials_download_url(
+            lambda: "test-access-token", "cid", "kubeconfig"
+        )
+        data = json.loads(resp)
+        assert data["url"] == "https://u/kubeconfig"
+
+
+@pytest.mark.asyncio
+async def test_tool_set_host_role_module() -> None:
+    from assisted_service_mcp.src.tools import host_tools
```

Comment:
Not a big deal but it's strange that these are imported with a different name in each test. I feel like that makes things more confusing, but :shrug: